pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -388,19 +397,65 @@ func (r *Reconciler) streamLogs(ctx context.Context, o results.Object, logType,
 		return fmt.Errorf("error reading from tkn reader: %w", err)
 	}
 
-	errChanRepeater := make(chan error)
-	go func(echan <-chan error, o metav1.Object) {
-		writeErr := <-echan
-		errChanRepeater <- writeErr
+	errChanRepeater := make(chan error, 100)
 
-		_, err := writer.Flush()
-		if err != nil {
-			logger.Error(err)
+	// logctx is derived from ctx. Therefore, if ctx is cancelled (either explicitly through a call to its cancel
+	// function or when it reaches its deadline), logctx will be cancelled automatically.
+	// TODO: Implement configurable timeout based on user feedback analysis.
+	logctx, _ := context.WithTimeout(ctx, 10*time.Minute)
+
+	go func(ctx context.Context, echan <-chan error, o metav1.Object) {
+		defer close(errChanRepeater)
+		select {
+		case <-ctx.Done():
+			logger.Warnw("Context done streaming log",
+				zap.String("namespace", o.GetNamespace()),
+				zap.String("name", o.GetName()),
+			)
+		case writeErr := <-echan:
```

Comment:
to close the loop @ramessesii2 on you comment https://github.com/tektoncd/results/pull/703/files#r1486049142 if you are only closing the channel in the go routine, what unblocks this call?

are we always hitting the `case <-ctx.Done()` path ?

Reply:
Hey Gabe, 
We don't hit ctx.Done() most of the time. Actually, I've never encountered that bcs of 10 min of timeout.

> case writeErr := <-echan, what unblocks this call?

My understanding is based on the fact that even if echan is closed by tkn reader, reading a closed channel, it will always be successful, returning the zero value.
We could run into problem if echan is a nil. Reading from or writing to a nil channel causes code to hang forever. But since the check was not there in the original Results code I assume we expect a non-nil channel from tknReader.
Although, in the next iteration, I'll address it.  I missed this edge case.

Reply:
neither the tekton reader or writer are closing this channel based on the code I've seen @ramessesii2 

can you provide the code link other than this goroutine where this channel is being closed?

Reply:
nevermind I found it :-)

Reply:
I think it is 

https://github.com/tektoncd/results/blob/main/vendor/github.com/tektoncd/cli/pkg/log/pipeline_reader.go#L50-L51

Reply:
sure, the flow is tknReader.read() -> readPipelineLog() -> finally landing to readAvailablePipelineLogs() for our case since we only stream longs once PipelineRuns/TRs have a specific closed state -> [close echan (errC)](https://github.com/tektoncd/results/blob/main/vendor/github.com/tektoncd/cli/pkg/log/pipeline_reader.go#L51)

Reply:
not sure how the original echan could ever  be nil, but in general defensive programming is good :-) 

I'll wait for you update and then we can go from there - thanks